maintenance/maintenanceTaskScheduler.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+require_once( dirname( __FILE__ ) . "/Maintenance.php" );
+
+class MaintenanceTaskScheduler extends Maintenance {
+	public function __construct() {
+		parent::__construct();
+
+		$this->mDescription = "Adds task execution to the queue";
+		$this->addArg( "script", "Script that is going to be executed", true );
+		$this->addOption( "active", "Run on wikis which are active at least <param> days" );
+		$this->addOption( "cluster", "Cluster name, possible values: c1, ..., c7" );
+		$this->addOption( "id", "Wiki IDs (separated with comma)" );
+	}
+
+	public function execute() {
+		global $wgExternalSharedDB;
+
+		$script = $this->getArg( 0 );
+		$active = $this->getOption( "active" );
+		$cluster = $this->getOption( "cluster" );
+
+		$idsParam = [];
+		if ( $this->getOption( "id" ) !== null ) {
+			$idsParam = array_map( "intval", explode( ",", $this->getOption( "id" ) ) );
+		}
+
+		$this->output( "Scheduling ${script}\n" );
+
+		$db = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		// TODO: Handle 300k wikis
+		$this->getWikis( $idsParam, $active, $cluster )
+			->runLoop( $db, function ( $data, $row ) use ( $script ) {
+				$this->scheduleTask( $row->city_id, $script );
+			} );
+	}
+
+	private function getWikis( array $idsParam = [], int $active = null, string $cluster = null ) {
+		$sql = ( new \WikiaSQL() )
+			->SELECT( "city_id" )
+			->FROM( "city_list" );
```

Comment:
Please add `city_public = 1` condition - we only want to process active wikis and skip closed ones.